Dh logging siebel (#455)
mulesoft/docs-connectors: modules/ROOT/pages/siebel/siebel-connector.adoc

> [NOTE]

Reviewer: HIDDEN TEXT ABOVE THE NOTE
edited here; look for changes

* 1 (ManagerView)
Displays records that a user and those who report to the user can access. For example, it includes the records that Siebel CRM displays in the My Team's Accounts visibility filter.

* 2 (PersonalView)
Displays records that a user can access, as determined by the Visibility Field property of the BusComp view mode object. For example, it includes the records that Siebel CRM displays in the My Accounts visibility filter.

* 3 (AllView)
Displays all records that include a valid owner. For example, it includes the records that Siebel CRM displays in the All Accounts Across Organizations visibility filter:

** Data Sense Filter Query Bus Objects
Use this field to write a query to filter the business objects metadata being downloaded into the application.
** DataSense Filter Query Bus Comp
Use this field to write a query to filter the business components metadata being downloaded into the application.
** DataSense Filter Query Int Objects
Use this field to write a query to filter the integration objects metadata being downloaded into the application.
** Data Sense Filter Query
Use this field to write a query to filter the business services metadata being downloaded into the application.
** Other
Use this field to define another query to filter the business objects, services, or integration objects metadata to be downloaded into the application.

> Limit the number of objects to retrieve through DataSense to a few objects using search specifications, otherwise retrieving metadata slows down Studio.

Reviewer: ```suggestion
Use search specifications to limit the number of objects to retrieve through DataSense to a few objects; otherwise, retrieving metadata slows down Studio.
```

[[log-requests-responses]]
== Log Requests and Responses 

To log requests and responses when using the connector, add the following line to the `Loggers` element of the `log4j2.xml` configuration file for the Mule app:

[source,xml,linenums]
----
<AsyncLogger name="org.mule.modules.siebel.internal.service"
   level="DEBUG"/>
----

> For example:

Reviewer: GLOBALLY avoid this brief fragment to introduce an example.
Either end the introductory text with a colon or make this fragment more descriptive, telling what this is an example of.

[source,xml,linenums]
----
<?xml version="1.0" encoding="UTF-8"?>
<Configuration status="WARN">
	<Appenders>
	    <Console name="Console" target="SYSTEM_OUT">
	    	<PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
		</Console>
	</Appenders>
	<Loggers>
		<Logger name="org.mule.modules.siebel.internal.service" level="debug" additivity="false">
			<appender-ref ref="Console" level="debug"/>
		</Logger>
	</Loggers>
</Configuration>
----

You can view the app log  as follows:

* If you’re running the app from the Anypoint Platform, the output is visible in the Anypoint Studio console window.

> * If you’re using Mule from the command line to run the app, the app log is visible in your OS console.

Reviewer: make parallel:
```suggestion
* If you’re running the app using Mule from the command line, the app log is visible in your OS console.
```

> You can also view the app log in this default location, unless the log file path was customized in the app’s log file (`log4j2.xml`):

Reviewer: ```suggestion
Unless the log file path was customized in the app’s log file (`log4j2.xml`), you can also view the app log in this default location:
```
To have the text directly introduce the location

`MULE_HOME/logs/<app-name>.log`

For more information about the app log, see xref:mule-runtime::logging-in-mule.adoc[Configuring Logging].

== Use Case: Studio

> The Siebel Connector is an operation-based connector,

Reviewer: Hidden TEXT

which means that when you add the connector to your flow, you must specify an operation to perform.

=== Example Use Cases
The following are common use cases for the Siebel v4.0.0 connector:
. Create a business component (Action, Account, and so on).
.. Create a new Mule project in Anypoint Studio.
.. Add a suitable Mule inbound endpoint, such as the HTTP listener, to begin the flow.
.. Drag the create business component operation from the Siebel JDB palette onto the canvas and then select it to open the properties editor.
.. Configure the operation's parameters:
+
image::siebel/siebel-studio-create.png[Configure the operation's parameters]
+
.. Drag the Transform Message connector between the HTTP Listener and the Siebel connector on the flow.
.. Configure the input payload with the desired fields:
+
image::siebel/siebel-studio-transform-create.png[Configure the input payload]
+

NOTE THAT this doc uses both "HTTP Listener" and "HTTP listener". Standardize to the correct presentation.

Reviewer: HIDDEN TEXT

. Execute a business service (Create Contact, Create Account, and so on).
+
.. Create a new Mule project in Anypoint Studio.
.. Add a suitable Mule inbound endpoint, such as the HTTP listener, to begin the flow.   
NOTE Listener vs. listener; search for and standardize GLOBALLY
.. Drag the execute business service operation from the Siebel JDB palette onto the canvas and then select it to open the properties editor.
.. Configure the operation's parameters:
+
image::siebel/siebel-execute.png[Configure the operation parameters]
+
.. Drag the Transform Message connector between the HTTP Listener and the Siebel connector on the flow.
.. Configure the input payload with the desired fields:
+
image::siebel/siebel-transform-execute.png[Configure the input payload]
+
. Execute Siebel adapter (creating an integration object)
+
.. Create a new Mule project in Anypoint Studio.
.. Add a suitable Mule inbound endpoint, such as the HTTP listener, to begin the flow.
.. Drag the create business component operation from the Siebel JDB palette onto the canvas and then select it to open the properties editor.
.. Configure the operation's parameters:

Reviewer: MORE HIDDEN TEXT

+
image::siebel/siebel-integration.png[Configure the operation parameters]
+
.. Drag the Transform Message connector between the HTTP Listener and the Siebel connector on the flow.
.. Configure the input payload with the desired fields:
+
image::siebel/siebel-transform-integration.png[Configure the input payload]

== Use Case: XML
CAN WE MAKE THIS A COMPLETE SENTENCE?
XML snippet from a project using the Siebel Connector:
[source,xml,linenums]
----
<?xml version="1.0" encoding="UTF-8"?>
<mule xmlns:ee="http://www.mulesoft.org/schema/mule/ee/core" xmlns:siebel="http://www.mulesoft.org/schema/mule/siebel"
      xmlns:http="http://www.mulesoft.org/schema/mule/http"
      xmlns="http://www.mulesoft.org/schema/mule/core" xmlns:doc="http://www.mulesoft.org/schema/mule/documentation"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mulesoft.org/schema/mule/core http://www.mulesoft.org/schema/mule/core/current/mule.xsd

== Log Requests and Responses [[log-requests-responses]]

> To log requests and responses when using the connector, add the following line to the `Loggers` element of the `log4j2.xml` configuration file for the Mule app:

Reviewer 2: How about showing an example of the Logger block?

Reviewer 3: Literally any log4j2.xml file of any connector or Mule application has that block. Here's an example file from the Service Bus connector for Mule 4:

```
<?xml version="1.0" encoding="UTF-8"?>
<Configuration status="WARN">
	<Appenders>
	    <Console name="Console" target="SYSTEM_OUT">
	    	<PatternLayout pattern="%d{HH:mm:ss.SSS} [%t] %-5level %logger{36} - %msg%n"/>
		</Console>
	</Appenders>
	<Loggers>
		<Logger name="org.mule.modules.microsoftservicebus" level="debug" additivity="false">
			<appender-ref ref="Console" level="debug"/>
		</Logger>
	</Loggers>
</Configuration>
```

[source,xml,linenums]
----
<AsyncLogger name="org.mule.modules.siebel.internal.service"
   level="DEBUG"/>
----

> You can view the app log  as follows:

Reviewer 2: Not sure what you mean by "app log" -- please explain in the text.

Author: This is wording from the Mule documentation.

* If you’re running the app from the Anypoint Platform, the output is visible in the Anypoint Studio console window.

* If you’re using Mule from the command line to run the app, the app log is visible in your OS console.

> You can also view the application log in this default location, unless the log file path was customized in the app’s log file (`log4j2.xml`):

Reviewer 2: Should use either all "application" or all "app" but not go back and forth.

Author: Agreed

		</Console>
	</Appenders>
	<Loggers>

> 		<AsyncLogger name="org.mule.modules.siebel.internal.service" level="DEBUG" additivity="FALSE">

Reviewer 3: `level="DEBUG"` is fine. But I haven't found evidence to confirm that `additivity="FALSE"` will, so I'd change that back to `false`.
